dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/common/jaxb/JAXBContextCache.java

```diff
@@ -0,0 +1,504 @@
+import java.io.InputStreamReader;
+import java.lang.ref.WeakReference;
+import java.lang.reflect.Method;
+import java.nio.charset.StandardCharsets;
+import java.security.AccessController;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.regex.Pattern;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.annotation.XmlElementDecl;
+import javax.xml.transform.dom.DOMSource;
+
+import org.apache.cxf.common.classloader.ClassLoaderUtils;
+import org.apache.cxf.common.util.CacheMap;
+import org.apache.cxf.common.util.CachedClass;
+import org.apache.cxf.common.util.StringUtils;
+import org.apache.cxf.common.logging.LogUtils;
+
+/**
+ *
+ */
+public final class JAXBContextCache {
+
+    /**
+     * Return holder of the context, classes, etc...
+     * Do NOT hold onto these strongly as that can lock the JAXBContext and Set<Class> objects
+     * into memory.  It preferred to grab the context and classes (if needed) from this object
+     * immediately after the call to getCachedContextAndSchemas and then discard it.  The
+     * main purpose of this class is to hold onto the context/set strongly until the caller
+     * has a chance to copy those into a place where they can hold onto it strongly as
+     * needed.
+     */
+    private static final Logger LOG = LogUtils.getLogger(JAXBContextCache.class);  // Liberty Change
+    public static final class CachedContextAndSchemas {
+        private final JAXBContext context;
+        private final Set<Class<?>> classes;
+        private final WeakReference<CachedContextAndSchemasInternal> ccas;
+        private CachedContextAndSchemas(JAXBContext context, Set<Class<?>> classes, CachedContextAndSchemasInternal i) {
+            this.context = context;
+            this.classes = classes;
+            ccas = new WeakReference<>(i);
+        }
+        public JAXBContext getContext() {
+            return context;
+        }
+        public Set<Class<?>> getClasses() {
+            return classes;
+        }
+        public Collection<DOMSource> getSchemas() {
+            CachedContextAndSchemasInternal i = ccas.get();
+            if (i != null) {
+                return i.getSchemas();
+            }
+            return null;
+        }
+
+        public void setSchemas(Collection<DOMSource> schemas) {
+            CachedContextAndSchemasInternal i = ccas.get();
+            if (i != null) {
+                i.setSchemas(schemas);
+            }
+        }
+
+    }
+    private static final class CachedContextAndSchemasInternal {
+
+        boolean isLoggableFine = LOG.isLoggable(Level.FINE);
+        boolean isLoggableInfo = LOG.isLoggable(Level.INFO); 
```

Comment:
these are no longer used and should be removed.

Reply:
Good catch, will remove